packages/teleport/src/lib/tdp/codec.ts

```diff
@@ -301,21 +322,15 @@ export default class Codec {
     throw new Error('Not implemented');
   }
 
-  // decodeMessageType decodes the MessageType from a raw tdp message
+  // _decodeMessageType decodes the MessageType from a raw tdp message
   // passed in as an ArrayBuffer (this typically would come from a websocket).
   // Throws an error on an invalid or unexpected MessageType value.
-  decodeMessageType(buffer: ArrayBuffer): MessageType {
+  _decodeMessageType(buffer: ArrayBuffer): MessageType {
     const messageType = new DataView(buffer).getUint8(0);
-    if (
-      messageType === MessageType.PNG_FRAME ||
-      messageType === MessageType.CLIPBOARD_DATA ||
-      messageType === MessageType.ERROR
-    ) {
-      return messageType;
-    } else {
-      // We don't expect to need to decode any other value on the client side
+    if (messageType > MessageType.ERROR) {
```

Comment:
This is okay for now but seems fragile as we add messages.

I might go with a switch and be explicit about the types you decode. This way it's obvious when one is missing.

Reply:
True, added a `TODO`.